File: src/main/java/eotree/data/EOCharData.java
package eotree.data;

import eotree.data.EOData;

/**
 * EBNF representation:
 * <code>
 * /'([^']|\\\d+)'/
 * </code>
 */
public class EOCharData extends EOData {
    public Character c;

    public EOCharData(Character c) {
        this.c = c;
    }

    @Override
    public String generateEO(int indent) {
        return "'" + c + "'";
    }
}

File: src/main/java/eotree/data/EOCharData.kt
package eotree.data

/**
 * EBNF representation:
 * `
 * /'([^']|\\\d+)'/
` *
 */
class EOCharData(var c: Char) : EOData() {
    override fun generateEO(indent: Int): String {
        return "'$c'"
    }
}
